cli/src/lib.rs

```diff
@@ -3036,9 +3046,16 @@ fn validator_flags(
             .map(|deployment| Ok(deployment.address.to_string()))
             .unwrap_or_else(|| program.pubkey().map(|p| p.to_string()))?;
 
-        flags.push("--bpf-program".to_string());
-        flags.push(address.clone());
-        flags.push(binary_path);
+        if use_upgradeable_program {
+            flags.push("--upgradeable-program".to_string());
+            flags.push(address.clone());
+            flags.push(binary_path);
+            flags.push(cfg.wallet_kp()?.pubkey().to_string());
+        } else {
+            flags.push("--bpf-program".to_string());
+            flags.push(address.clone());
+            flags.push(binary_path);
+        }
 
```

Comment:
We also deploy programs similarly with genesis configuration here https://github.com/coral-xyz/anchor/blob/d1e32674d50005d04160b6fbac6f91de602a8829/cli/src/lib.rs#L3056-L3069

Can add a `upgradeable: bool` field to `GenesisEntry` https://github.com/coral-xyz/anchor/blob/d1e32674d50005d04160b6fbac6f91de602a8829/cli/src/config.rs#L947-L952

which would allow us to specify

```toml
[[test.genesis]]
address = "Test111111111111111111111111111111111111111"
program = "test.so"
upgradeable = true
```

Reply:
done!  added `upgradeable: bool` to `GenesisEntry`